src/System.Console/tests/SyncTextReader.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class SyncTextReaderTests
+{
+    // NOTE: These tests test the underlying SyncTextReader by
+    // accessing the Console.In TextReader (which in fact is a SyncTextReader).
+
+    static readonly string[] s_testLines = new string[] {
```

Comment:
To be honest I was not sure of whether or not the content in s_testLines was significant or not, so since I'm currently cargo culting this code base, I simply chose to use the same test data for other similar methods that was tested. :smile:

I will leave it as it is for now, and revisit it later on. Making sure that the test data covers the scenarios you mention make sense to me.